File: core/src/net/brenig/pixelescape/render/overlay/EmptyOverlay.java
package net.brenig.pixelescape.render.overlay;

import net.brenig.pixelescape.game.data.GameDebugSettings;
import net.brenig.pixelescape.screen.GameScreen;

/**
 * Empty Overlay used to avoid null in GameScreen
 */
public final class EmptyOverlay extends Overlay {

	public EmptyOverlay(final GameScreen screen) {
		super(screen);
	}

	@Override
	public void show() {
		screen.resetInputManager();
	}

	@Override
	public void render(float delta) {
	}

	@Override
	public void pause() {
		screen.showGamePausedOverlay();
	}

	@Override
	public void resume() {
		screen.showGamePausedOverlay();
	}

	@Override
	public void onResize(int width, int height) {
		if (screen.isInitialized() && GameDebugSettings.Companion.get("AUTO_PAUSE")) {
			screen.showGamePausedOverlay();
		}
	}

	@Override
	public boolean shouldHideGameUI() {
		return false;
	}

	@Override
	public boolean shouldPauseOnEscape() {
		return true;
	}
}

File: core/src/main/kotlin/net/brenig/pixelescape/render/overlay/EmptyOverlay.kt
package net.brenig.pixelescape.render.overlay

import net.brenig.pixelescape.game.data.GameDebugSettings
import net.brenig.pixelescape.screen.GameScreen

/**
 * Empty Overlay used to avoid null in GameScreen
 */
class EmptyOverlay(screen: GameScreen) : Overlay(screen) {

    override fun show() {
        screen.resetInputManager()
    }

    override fun render(delta: Float) {}

    override fun pause() {
        screen.showGamePausedOverlay()
    }

    override fun resume() {
        screen.showGamePausedOverlay()
    }

    override fun onResize(width: Int, height: Int) {
        if (screen.isInitialized && GameDebugSettings["AUTO_PAUSE"]) {
            screen.showGamePausedOverlay()
        }
    }

    override fun shouldHideGameUI(): Boolean {
        return false
    }

    override fun shouldPauseOnEscape(): Boolean {
        return true
    }
}
